posts/2024-08-09-Project-goals.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: post
+title: "Rust Project goals for 2024"
+author: Niko Matsakis
+team: Leadership Council <https://www.rust-lang.org/governance/teams/leadership-council>
+---
+
+With the merging of [RFC #3672][], the Rust project has selected a **slate of 26 Project Goals** for the second half of 2024 (2024H2). This is our first time running an [experimental new roadmapping process][RFC #3614]; we expect to be running the process roughly every six months. Of these goals, we have designated three of them as our **flagship goals**, representing our most ambitious and most impactful efforts: (1) finalize preparations for the Rust 2024 edition; (2) bring the Async Rust experience closer to parity with sync Rust; and (3) resolve the biggest blockers to the Linux kernel building on stable Rust. As the year progresses we'll be posting regular updates on these 3 flagship goals along with the 23 others.
+
+[RFC #3672]: https://github.com/rust-lang/rfcs/pull/3672#issuecomment-2254599176
+[RFC #3614]: https://github.com/rust-lang/rfcs/pull/3614
+
+## Rust’s mission
+
+All the goals selected ultimately further Rust's mission of **empowering everyone to build reliable and efficient software**. Rust targets programs that prioritize
+
+* reliability and robustness;
+* performance, memory usage, and resource consumption; and
+* long-term maintenance and extensibility.
```

Comment:
Are these actually from somewhere? Or is this list something you've iterated on?

Reply:
We don't have anything official, it's the result of iteration by me over time (though I do plan to open an RFC along these lines...).

Reply:
It was in the RFC, though.

Reply:
So I guess the answer is, yes, they are from somewhere: the project goals RFC. =)

Reply:
I suppose this is true. Just wanted to double check, because it feels "big" in the sense of direction+alignment of the Project.

Given that the Project Goals RFC had broad (but not complete, e.g. no infra checkboxes) buy-in, I'm inclined to say I think this is good to include. That being said, I'd love to see a separate RFC more directed to this!

Reply:
Agreed on all counts